sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py

```diff
@@ -423,10 +432,18 @@ def refresh(self, **kwargs) -> None:
                             key,
                             label,
                         )
-                        self._load_all(**kwargs)
-                        self._refresh_on[(key, label)] = updated_sentinel.etag
-                        self._refresh_timer.reset()
-                        return
+                        need_refresh = True
+
+                        updated_sentinel_keys[(key, label)] = updated_sentinel.etag
+                # Need to only update once, no matter how many sentinels are updated
+                if need_refresh:
+                    self._load_all(**kwargs)
+                    self._refresh_on = updated_sentinel_keys
```

Comment:
Its possible that sentinel keys have been updated again after we already stored their etags in `updated_sentinel_keys`. So after `_load_all`, configuration would have the updated sentinel values, but we are still storing old etags that we'll use for the next round of refresh.

Reply:
Or we could pass the already loaded sentinel keys into the _load_all method.

Reply:
This should be fixed.